dattri/model_utils/hook.py

```diff
@@ -8,30 +9,35 @@
 if TYPE_CHECKING:
     from typing import List, Tuple
 
+    from torch.utils.data import DataLoader
+
 import torch
 from torch import Tensor
 
 
-def get_layer_feature(
+def _get_layer_feature(
```

Comment:
Sorry for the confusion. Since now I assume that the user must input a model with a final linear layer, the first one will be deprecated. (The second one assumes the model to have a final linear layer and directly get the input/output of that layer, which will be inputs of RPS)

Reply:
Will remove `_get_layer_feature` and add some words in the docstring of `get_final_layer_io` and `RPSAttributor`.